vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc

```diff
@@ -23,56 +23,39 @@ using std::placeholders::_1;
 using std::placeholders::_2;
 using std::placeholders::_3;
 
-namespace braveledger_wallet {
-
-Create::Create(bat_ledger::LedgerImpl* ledger) :
-    ledger_(ledger) {
-  initAnonize();
-}
-
-Create::~Create() {
-}
-
-void Create::Start(ledger::ResultCallback callback) {
-  auto req_callback = std::bind(&Create::RequestCredentialsCallback,
-      this,
-      _1,
-      std::move(callback));
+namespace {
 
-  const std::string url =
-      braveledger_request_util::BuildUrl(REGISTER_PERSONA, PREFIX_V2);
-
-  ledger_->LoadURL(url, {}, "", "", ledger::UrlMethod::GET, req_callback);
-}
-
-std::string Create::GetAnonizeProof(const std::string& registrarVK,
-                                    const std::string& id,
-                                    std::string* preFlight) {
+std::string GetAnonizeProof(
+    const std::string& registrarVK,
+    const std::string& id,
+    std::string* preFlight) {
   const char* cred = makeCred(id.c_str());
-  if (cred != nullptr) {
-    *preFlight = cred;
-    // should fix in
-    // https://github.com/brave-intl/bat-native-anonize/issues/11
-    free((void*)cred); // NOLINT
-  } else {
+  if (cred == nullptr) {
     return "";
   }
-  const char* proofTemp = registerUserMessage(preFlight->c_str(),
-                                              registrarVK.c_str());
-  std::string proof;
-  if (proofTemp != nullptr) {
-    proof = proofTemp;
-    // should fix in
-    // https://github.com/brave-intl/bat-native-anonize/issues/11
-    free((void*)proofTemp); // NOLINT
-  } else {
+
+  *preFlight = cred;
+  // should fix in
+  // https://github.com/brave-intl/bat-native-anonize/issues/11
+  free((void*)cred); // NOLINT
```

Comment:
Just double-checking: this was allocated with `malloc` and not `new`?

Reply:
not sure about it as code was just moved, no code/logic change

Reply:
No worries, I checked and the memory is allocated via `srtdup` so `free` is correct.